src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,725 @@
+    **)
+  let get_coeff_vec (t: t) texp =
+    let open Apron.Texpr1 in
+    let exception NotLinearExpr in
+    let exception NotIntegerOffset in
+    let negate coeff_var_list = List.map (fun (coeff, var) -> (Z.(-coeff), var)) coeff_var_list in
+    let multiply_with_Z number coeff_var_list =
+      List.map (fun (coeff, var) -> (Z.(number * coeff, var))) coeff_var_list in
+    let multiply a b =
+      (* if one of them is a constant, then multiply. Otherwise, the expression is not linear*)
+      match a, b with
+      | [(a_coeff, None)], b -> multiply_with_Z a_coeff b
+      | a, [(b_coeff, None)] -> multiply_with_Z b_coeff a
+      | _ -> raise NotLinearExpr
+    in
+    let rec convert_texpr texp =
+      begin match texp with
+        (*If x is a constant, replace it with its const. val. immediately*)
+        | Cst (Interval _) ->failwith "Not a constant"
+        | Cst (Scalar x) ->
+          begin match SharedFunctions.int_of_scalar ?round:None x with
+            | Some x -> [(x, None)]
+            | None -> raise NotIntegerOffset end
+        | Var x ->
+          let var_dim = Environment.dim_of_var t.env x in
+          begin match t.d with
+            | None -> [(Z.one, Some var_dim)]
+            | Some d ->
+              (if Option.is_some (fst d.(var_dim)) then [(Z.one, fst d.(var_dim))]
+               else [])
+              @ [(snd d.(var_dim), None)]
+          end
+        | Unop (u, e, _, _) ->
+          begin match u with
+            | Neg -> negate (convert_texpr e)
+            | Cast -> convert_texpr e (*Ignore since casts in apron are used for floating point nums and rounding in contrast to CIL casts*)
+            | Sqrt -> raise NotLinearExpr end
+        | Binop (b, e1, e2, _, _) ->
+          begin match b with
+            | Add -> List.concat [convert_texpr e1; convert_texpr e2]
+            | Sub -> List.concat [convert_texpr e1; negate (convert_texpr e2)]
+            | Mul -> multiply (convert_texpr e1) (convert_texpr e2)
+            | _ -> raise NotLinearExpr end
+      end
+    in match convert_texpr texp with
+    | exception NotLinearExpr -> None
+    | x -> Some(x)
+
+  let get_coeff (t: t) texp =
+    let d = Option.get t.d in
+    match get_coeff_vec t texp with
+    | None -> None (*The (in-) equality is not linear, therefore we don't know anything about it. *)
+    | Some cv's ->
+      let expr = Array.make (Environment.size t.env) Z.zero in
+      let accumulate_constants a (c, v) = match v with
+        | None -> Z.(a + c)
+        | Some idx -> let (term,con) = d.(idx) in
+          (Option.may (fun ter -> expr.(ter) <- Z.(expr.(ter) + c)) term;
+           Z.(a + c * con))
+      in
+      let constant = List.fold_left accumulate_constants Z.zero cv's in (* abstract simplification of the guard wrt. reference variables *)
+      let sum_of_terms = Array.fold_lefti (fun list v (c) -> if Z.equal c Z.zero then list else (c,v)::list) [] expr in
+      match sum_of_terms with
```

Comment:
This structure is pretty much the same as in meet_tcons -> maybe we need another helper in front of get_coeff_vec